staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/validation/validation.go

```diff
@@ -42,5 +42,36 @@ func ValidateListOptions(options *internalversion.ListOptions) field.ErrorList {
 			allErrs = append(allErrs, field.Forbidden(field.NewPath("resourceVersionMatch"), "resourceVersionMatch \"exact\" is forbidden for resourceVersion \"0\""))
 		}
 	}
+	if options.SendInitialEvents {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("sendInitialEvents"), "sendInitialEvents is forbidden for list"))
+	}
+	return allErrs
+}
+
+func validateWatchOptions(options *internalversion.ListOptions, isWatchListFeatureEnabled bool) field.ErrorList {
+	allErrs := field.ErrorList{}
+	match := options.ResourceVersionMatch
+	if options.SendInitialEvents {
+		if len(match) == 0 {
```

Comment:
We can check more - if SendInitialEvents is set, then we know that match should exactly be `NotOlderThan`, right?

Reply:
This comment still holds - please fix.

Reply:
That would require even more branches. Plus deduplication logic. We do check the value of `ResourceVersionMatch` a few lines below.

Reply:
Why - if SendInitialEvents is set, the only option we support is actuall `NotOlderThan` nothing more.